File: Tests/server/server_executors/executors/ChunksExecutor.cpp
#include "ChunksExecutor.h"

#include "TextGenerator.h"

void ChunksExecutor::doGet(framework::HTTPRequest& request, framework::HTTPResponse& response)
{
	request.sendChunks<TextGenerator>(response, "Some information here");
}

void ChunksExecutor::doPost(framework::HTTPRequest& request, framework::HTTPResponse& response)
{
	request.sendFileChunks<TextGenerator>(response, "file.txt", "Some information here");
}

DECLARE_EXECUTOR(ChunksExecutor)

File: Tests/executors/CSharp/CoreExecutors/ChunksExecutor.cs
using CoreExecutors;
using Framework;

public class ChunksExecutor : HeavyOperationStatelessExecutor
{
	public override void DoGet(HttpRequest request, HttpResponse response)
	{
		TextGenerator textGenerator = new("Some information here");

		request.SendChunks(response, textGenerator);
	}

	public override void DoPost(HttpRequest request, HttpResponse response)
	{
		TextGenerator textGenerator = new("Some information here");

		request.SendFileChunks(response, textGenerator, "file.txt");
	}
}
